addons/test_sale_product_configurators/tests/test_sale_product_configurator.py

```diff
@@ -1,48 +1,42 @@
-class TestUi(odoo.tests.HttpCase, TestProductConfiguratorCommon):
-
-    def setUp(self):
-        super(TestUi, self).setUp()
-        self.custom_pricelist = self.env['product.pricelist'].create({
-            'name': 'Custom pricelist (TEST)',
-            'item_ids': [(0, 0, {
-                'base': 'list_price',
-                'applied_on': '1_product',
-                'product_tmpl_id': self.product_product_custo_desk.id,
-                'price_discount': 20,
-                'min_quantity': 2,
-                'compute_price': 'formula'
-            })]
-        })
+from odoo.tests.common import HttpCase, tagged
+
+from odoo.addons.mail.tests.common import mail_new_test_user
+from odoo.addons.sale_product_configurator.tests.common import TestProductConfiguratorCommon
+
+
+@tagged('post_install', '-at_install')
+class TestProductConfiguratorUi(HttpCase, TestProductConfiguratorCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Adding sale users to test the access rights
+        cls.salesman = mail_new_test_user(
+            cls.env,
+            name='Salesman',
+            login='salesman',
+            password='salesman',
+            groups='sales_team.group_sale_salesman',
+        )
+
+        # Setup partner since user salesman don't have the right to create it on the fly
+        cls.env['res.partner'].create({'name': 'Tajine Saucisse'})
 
     def test_01_product_configurator(self):
-        # To be able to test the product configurator, admin user must have access to "variants" feature, so we give him the right group for that
-        self.env.ref('base.user_admin').write({'groups_id': [(4, self.env.ref('product.group_product_variant').id)]})
-        self.start_tour("/web", 'sale_product_configurator_tour', login="admin")
+        self.start_tour("/web", 'sale_product_configurator_tour', login='salesman')
 
     def test_02_product_configurator_advanced(self):
-        # group_product_variant: use the product configurator
-        # group_sale_pricelist: display the pricelist to determine when it is changed after choosing
         # group_delivery_invoice_address: show the shipping address (needed for a trigger)
-        #                       the partner
-        self.env.ref('base.user_admin').write({
-            'groups_id': [
-                (4, self.env.ref('product.group_product_variant').id),
-                (4, self.env.ref('product.group_product_pricelist').id),
-                (4, self.env.ref('account.group_delivery_invoice_address').id),
-            ],
+        self.salesman.sudo().write({
+            'groups_id': [(4, self.env.ref('account.group_delivery_invoice_address').id)],
         })
 
         # Prepare relevant test data
         # This is not included in demo data to avoid useless noise
-        product_attributes = self.env['product.attribute'].create([{
+        product_attributes = self.env['product.attribute'].sudo().create([{
```

Comment:
is the sudo still needed now that you don't use @users('salesman') ?

(Same comment for all the sudos you are adding)

Reply:
Nop, this was an overlooked when taking off the @users decorator, my bad =]